api/src/main/java/org/apache/iceberg/expressions/ExpressionVisitors.java

```diff
@@ -51,6 +53,10 @@ public R or(R leftResult, R rightResult) {
       return null;
     }
 
+    public <T> R predicate(BoundSetPredicate<T> pred) {
+      return null;
```

Comment:
This should also throw `UnsupportedOperationException` to fail any implementation that is missing support.